### PR TITLE
Fixed the case of risk models with different curve resolutions

### PR DESCRIPTION
**Pull request by micheles**
Fixes https://github.com/gem/oq-risklib/issues/632. I have checked the numbers with the ones produced by the engine calculator in release 1.7, and they match. Anirudh is double-checking them with an independent implementation. This PR must enter together with https://github.com/gem/oq-engine/pull/1921.

All of Anirudh's test are running except case_master that will be fixed later, when the insured losses will be computed again.

### CODE REVIEW

**Comment by raoanirudh** [comment]
LGTM

File: openquake/calculators/base.py [hunk 1]
```diff
@@ -54,6 +54,19 @@ class AssetSiteAssociationError(Exception):
 rlz_dt = numpy.dtype([('uid', (bytes, 200)), ('weight', float)])
 
 
+def set_array(longarray, shortarray):
+    """
+    :param longarray: an array of length L >= l
+    :param shortarray: an array of length l
```
Comment (ptormene) [inline openquake/calculators/base.py]: Please specify in the docstring that those should be numpy arrays. For lists, for instance, it breaks.

Comment (ptormene) [inline openquake/calculators/base.py]: And for numpy arrays, for some reason I get numbers instead of nan

**Comment by ptormene** [comment]
Please double-check about the `set_array` function. Apart from that, LGTM

File: openquake/calculators/base.py (hunk 1 above)
Comment (micheles) [inline openquake/calculators/base.py]: This happens if you use arrays of integers instead of array of floats

**Comment by ptormene** [comment]
LGTM

**Pull request merged**